scripts/Palindrome.py

```diff
@@ -1,41 +1,40 @@
+IGNORE_CHAR_LIST = ' .,-_:!@#$%^&?'
+
+
 def is_palindrome(sentence):
-    value = True
-    len_s = len(sentence) - 1
-    half_len = len(sentence) // 2
+    result = True
     left_side_idx = 0
-    right_side_idx = 0
-    valid_char_count = 0
-    ignore_char_count = 0
+    right_side_idx = len(sentence) - 1
 
-    while valid_char_count // 2 <= half_len - ignore_char_count // 2:
+    while left_side_idx != right_side_idx:
         left_side_element = sentence[left_side_idx]
-        right_side_element = sentence[len_s - right_side_idx]
-        if left_side_element not in IGNORE_CHAR_LIST and right_side_element not in IGNORE_CHAR_LIST:
+        right_side_element = sentence[right_side_idx]
+        left_side_letter = left_side_element not in IGNORE_CHAR_LIST
+        right_side_letter = right_side_element not in IGNORE_CHAR_LIST
+        if left_side_letter and right_side_letter:
             left_side_idx += 1
-            right_side_idx += 1
-            valid_char_count += 1
-            if right_side_element != left_side_element:
-                value = False
+            right_side_idx -= 1
+            if left_side_element != right_side_element:
+                result = False
                 break
```

Comment:
Replace **break** (that is not a best practices approach) with the alternative solution **WITHOUT** break / continue